glean-core/csharp/Glean/Glean.cs

```diff
@@ -204,13 +213,9 @@ public void SetUploadEnabled(bool enabled)
 
                 if (!enabled)
                 {
-                    // Cancel any pending workers here so that we don't accidentally upload or
+                    // Cancel any pending tasks here so that we don't accidentally upload or
                     // collect data after the upload has been disabled.
-                    // TODO: metricsPingScheduler.cancel()
-
-                    // Cancel any pending workers here so that we don't accidentally upload
-                    // data after the upload has been disabled.
-                    // TODO: PingUploadWorker.cancel(applicationContext)
+                    Dispatchers.CancelBackgroundTasks();
```

Comment:
Let's leave this as it is for now. This will be taken care in the future, when the metrics ping scheduler is there.

Reply:
FWIW, just calling this shouldn't hurt anything since the metrics ping scheduler will likely be using the ConcurrentScheduler to run the upload task on, it would be canceled by this function anyway.

Reply:
Probably, but let's take this to when we'll implement these pieces ;)